refactron/analysis/symbol_table.py

```
@@ -120,22 +138,79 @@ def __init__(self, cache_dir: Optional[Path] = None):
+            # Fallback for old cache entries without hashes
+            return stat.st_mtime != metadata.get("mtime")
+        except Exception:
+            return True
+
+        return False
+
+    def _calculate_hash(self, file_path: Path) -> str:
+        """Calculate SHA-256 hash of file content."""
+        try:
+            return hashlib.sha256(file_path.read_bytes()).hexdigest()
+        except Exception:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)